src/FeatureDiscovery.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+class FeatureDiscovery extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {mounted: false};
+    this.open = this.open.bind(this);
+    this.close = this.close.bind(this);
+    this.close = this.close.bind(this);
+    this.reference = this.reference.bind(this);
+  }
+
+  componentDidMount() {
+    this.setState({mounted: true});
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (!this.props.open && nextProps.open) {
+      this.open();
+    } else if (nextProps.open === false) {
+      this.close();
+    }
+  }
+
+  reference(element) {
+    this._element = element;
+    if (this.props.open) {
+      this.open();
+    }
```

Comment:
I would rather move this logic to `componentDidMount` like other components do:
https://github.com/react-materialize/react-materialize/blob/188ee6c4521a1afc16548108cde2637d9bdc8ce6/src/Parallax.js#L7

Reply:
Same as adding a ref to a node: https://github.com/react-materialize/react-materialize/blob/188ee6c4521a1afc16548108cde2637d9bdc8ce6/src/Tabs.js#L40

Reply:
Then we wouldn't need this method `reference` which name is also misleading :)

Reply:
The reason why that's there is because the _tap target_ is not rendered until `mounted` is set to `true`.
This happens on `componentDidMount` because, as stated earlier, it's at this stage when we can assure
that the button we're targeting has been rendered already. Then, because `setState` is called, the component
is re-rendered and, because `mounted` is `true`, the actual _tap target_ is now rendered instead of an empty `div`.

Finally then we want to initialize the tap target accordingly once it's created, which is what happens within
the `reference` method.

Basically the issue here is that we cannot create the tap target before our target button is rendered/created
because the materialize library expects it to exist.

Reply:
Could we solve this by passing the target as a prop, like with Modal?

Reply:
Maybe an issue is that fab button never receives this ID
https://github.com/react-materialize/react-materialize/blob/master/src/Button.js#L85

Reply:
If I add a debugger in `componentDidMount` I can see that the target button is rendered

Reply:
Could be a solution, I guess. I do like the idea of being able to specify an ID as the target though, instead of an element. We could perhaps support both?